src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1272,6 +1272,73 @@ object ICustomAstVisitor.VisitFileRedirection(FileRedirectionAst fileRedirection
 
         private void InferTypesFrom(CommandAst commandAst, List<PSTypeName> inferredTypes)
         {
+            if (commandAst.Redirections.Count > 0)
+            {
+                var mergedStreams = new HashSet<RedirectionStream>();
+                bool allStreamsMerged = false;
+                foreach (RedirectionAst streamRedirection in commandAst.Redirections)
+                {
+                    if (streamRedirection is FileRedirectionAst fileRedirection)
+                    {
+                        if (fileRedirection.FromStream is RedirectionStream.All or RedirectionStream.Output)
+                        {
+                            // command output is redirected so it returns nothing.
+                            return;
+                        }
+                    }
+                    else if (streamRedirection is MergingRedirectionAst mergeRedirection && mergeRedirection.ToStream == RedirectionStream.Output)
+                    {
+                        if (mergeRedirection.FromStream == RedirectionStream.All)
+                        {
+                            allStreamsMerged = true;
+                            continue;
```

Comment:
Why do we need to continue? Can we break or fast return? Or follow possible `Get-PSDrive *>&1 4>&2`?

Reply:
This is possible: `ls C:\, L:\ *>&1 1>$HOME\MyData.txt` First you redirect everything to the success stream, and then you redirect the success stream to a file.